_posts/2021-10-25-waku-v1-vs-waku-v2.md

```diff
@@ -0,0 +1,202 @@
+The local Waku v2 simulation maintains the message throughput rate without any problems.
+
+![](/assets/img/waku1-vs-waku2/waku1-vs-waku2-85-nodes.png)
+
+### Iteration 5: 150 nodes
+
+Finally, we simulate message propagation in a network of 150 nodes.
+Due to local resource constraints, we run this simulation at a lower rate -
+35 messages per second -
+and for a shorter amount of time. 
+
+![](/assets/img/waku1-vs-waku2/waku1-vs-waku2-150-nodes.png)
+
+Notice how the Waku v1 bandwidth usage is now more than 10 times worse than that of Waku v2.
+This is to be expected, as each Waku v1 node will try to flood each new message to 149 other peers,
+while the Waku v2 nodes limit their full-message peerings to no more than 12.
+
+### Discussion
+
+Let's summarize average bandwidth growth against network growth for a constant message propagation rate.
+Since we are particularly interested in how Waku v1 compares to Waku v2 in terms of bandwidth usage,
+the results are normalised to the Waku v2 average bandwidth usage for each network size.
+
+![](/assets/img/waku1-vs-waku2/waku1-vs-waku2-overall-network-size.png)
+
+Extrapolation is a dangerous game,
+but it's safe to deduce that the divergence will only grow for even larger network topologies.
+Although control signalling contributes more towards overall bandwidth for Waku v2 networks,
+this effect becomes less noticeable for larger networks.
+For network segments with more than ~18 densely connected nodes,
+the advantage of using Waku v2 above Waku v1 becomes clear.
+
+## Network traffic comparison
+
+The analysis above controls the average message rate while network size grows.
+In reality, however, active users (and therefore message rates) are likely to grow in conjunction with the network.
+This will have an effect on bandwidth for both Waku v1 and Waku v2, though not in equal measure.
+Consider the impact of an increasing rate of messages in a network of constant size:
+
+![](/assets/img/waku1-vs-waku2/waku1-vs-waku2-overall-message-rate.png)
+
+The _rate_ of increase in bandwidth for Waku v2 is slower than that for Waku v1 for a corresponding increase in message propagation rate.
+In fact, for a network of 30 densely-connected nodes,
+if the message propagation rate increases by 1 per second,
+Waku v1 requires an increased average bandwidth of almost 70kbps at each node.
+A similar traffic increase in Waku v2 requires on average 40kbps more bandwidth per peer, just over half that of Waku v1.
+
+## Conclusions
+
+- **Waku v2 scales significantly better than Waku v1 in terms of average bandwidth usage**,
+especially for densely connected networks.
+- E.g. for a network consisting of **150** or more densely connected nodes,
+Waku v2 provides more than **10x** better average bandwidth usage rates than Waku v1.
+- As the network continues to scale, both in absolute terms (number of nodes) and in network traffic (message rates) the disparity between Waku v2 and Waku v1 becomes even larger.
+
```

Comment:
Future work section or paragraph perhaps hinting at next steps here? Make the post a bit more "dynamic"

(should've caught this earlier review)

Reply:
Added future work section in 61e5fec and re-requested review mostly for this. Also added other suggested changes to intro and moved the date forward again.

Reply:
Good stuff!